types/src/v2/params.rs

```diff
@@ -482,4 +506,26 @@ mod test {
 		assert_eq!(dsr.subscription, SubscriptionId::Str("9".into()));
 		assert_eq!(dsr.result, serde_json::json!("offside"));
 	}
+
+	#[test]
+	fn params_sequence_optional_ignore_empty() {
+		let params = RpcParams::new(Some(r#"["foo", "bar"]"#));
+		let mut seq = params.sequence();
+
+		assert_eq!(seq.optional_next_ignore_empty_json::<&str>().unwrap(), Some("foo"));
+		assert_eq!(seq.optional_next_ignore_empty_json::<&str>().unwrap(), Some("bar"));
+
+		let params = RpcParams::new(Some(r#"[]"#));
+		let mut seq = params.sequence();
+		assert_eq!(seq.optional_next_ignore_empty_json::<&str>().unwrap(), None);
+
+		let params = RpcParams::new(Some(r#"[12, "[]"]"#));
+		let mut seq = params.sequence();
+		assert_eq!(seq.optional_next_ignore_empty_json::<u64>().unwrap(), Some(12));
+		assert_eq!(seq.optional_next_ignore_empty_json::<&str>().unwrap(), Some("[]"));
+
+		let params = RpcParams::new(Some(r#"{}"#));
+		let mut seq = params.sequence();
+		assert_eq!(seq.optional_next_ignore_empty_json::<&str>().unwrap(), None);
+	}
```

Comment:
I added a test here while playing with this, not sure if it makes sense but here it is, your call. :
```suggestion
	}

	#[test]
	fn parse_optional_nested_params() {
		let nested = RpcParams::new(Some(r#"[1, [2], [3, 4], [[5], [6, 7], []]"#));
		let mut seq = nested.sequence();
		assert_eq!(seq.optional_next_ignore_empty_json::<i8>().unwrap(), Some(1));
		assert_eq!(seq.optional_next_ignore_empty_json::<[i8; 1]>().unwrap(), Some([2]));
		assert_eq!(seq.optional_next_ignore_empty_json::<Vec<u16>>().unwrap(), Some(vec![3, 4]));
		assert_eq!(seq.optional_next_ignore_empty_json::<Vec<Vec<u32>>>().unwrap(), Some(vec![vec![5], vec![6, 7], vec![]]));
	}	
	
```